ios/REANodesManager.mm

```diff
@@ -173,6 +173,18 @@ @implementation REANodesManager {
 #endif
 }
 
+- (CADisplayLink *)getDisplayLink
+{
+  RCTAssertMainQueue();
```

Comment:
It seems that the issue will still occur with this approach, since the code is not being dispatched to the main queue. Or am I missing something?

![Simulator Screenshot - iPhone 14 Pro - 2023-07-19 at 22 29 24](https://github.com/software-mansion/react-native-reanimated/assets/88153729/db1d8182-c48e-460e-a207-4987223dbb31)

This assertion will fail when running from a non-main thread. 
![Screenshot 2023-07-19 at 10 41 37 PM](https://github.com/software-mansion/react-native-reanimated/assets/88153729/8b0002c8-4005-418f-8889-958ec813d870)